airbyte-workers/src/main/java/io/airbyte/workers/internal/AirbyteMessageTracker.java

```diff
@@ -386,4 +416,54 @@ protected Long calculateMean(final Long currentMean, final Long totalCount, fina
     return (long) result;
   }
 
+  void addStateMessageToStreamToStateHashTimestampTracker(final AirbyteStateMessage stateMessage, final int stateHash, final DateTime timeEmitted) {
+    final List<StreamDescriptor> streamDescriptorsToUpdate = StateMessageHelper.getStreamDescriptors(stateMessage);
+
+    streamDescriptorsToUpdate.forEach(streamDescriptor -> {
+      final HashMap<Integer, DateTime> stateHashToTimestamp = new HashMap<>();
+      stateHashToTimestamp.put(stateHash, timeEmitted);
+      streamDescriptorToStateMessageTimestamps.put(streamDescriptor, stateHashToTimestamp);
```

Comment:
Aren't we erasing the previous data of the streamDescriptor in this loop? It should probably be a create if not exist instead.

Reply:
yes! thank you